Added note about createMiddleware helper function

btroncone opened the pull request:
I added a note and example for the new createMiddleware helper function. I also added a warning before the basic middleware example reminding developers to import the correct RxJS operators. There have been 2 issues opened and multiple questions in chat regarding this problem within the past week.

Long term I am going to start working on more comprehensive documentation for the github.io site. This will help new users hit the ground running as this project continues to grow in popularity!

btroncone committed:
Added note about createMiddleware helper function

I added a note and example for the new createMiddleware helper function. I also added a warning before the basic middleware example reminding developers to import the correct RxJS operators. There have been 2 issues opened and multiple questions in chat regarding this problem within the past week.

Long term I am going to start working on more comprehensive documentation for the github.io site. This will help new users hit the ground running as this project continues to grow in popularity!

MikeRyanDev commented:
Can you also add an example that shows passing in middleware factory providers to `usePreMiddleware` and/or `usePostMiddleware` ?

btroncone commented:
Yep. Also, do you think the thunks example is too complex for basic docs? I went back and forth but don't want to confuse anyone.

MikeRyanDev commented:
Probably too complex. Here's the same middleware but slightly simplified:

``` ts
const thunk = createMiddleware(function(dispatcher: Dispatcher<Action>) {
  return function(all$: Observable<Action | Thunk>){
    const [thunks$, actions$] = all$.partition(t => typeof t === 'function');

    thunks$.forEach(thunk => thunk(action => dispatcher.dispatch(action));

    return actions$;
  }
}, [Dispatcher]);
```

MikeRyanDev commented:
You should probably go ahead and stop pointing to store-saga as an example. It is about to become a lot more complex in the coming days.

btroncone committed:
updated createMiddleware example

I updated the thunk example to the simplified version and added an example initialization with usePreMiddleware. I also removed the reference to ngrx-saga middleware.

btroncone commented:
No worries, I removed it and added an example passing in factory provider to `usePreMiddleware`.

MikeRyanDev commented:
Awesome, lgtm :+1:
